ui/src/main/js/model/rest-api.js

```diff
@@ -28,21 +28,63 @@ var ajax = require('../util/ajax');
  * Workflow REST API
  */
 
+var cache = {};
+
+function addCacheEntry (key, ob) {
+    if (typeof ob === 'object') {
+        if (ob.status && (ob.status === 'SUCCESS' || ob.status === 'ABORTED' || ob.status === 'FAILED' || ob.status === 'UNSTABLE')) {
```

Comment:
:bug: why just these, and not, say, `NOT_BUILT` (used for example by `milestone`)?